backend/service/src/drug_interactions/tests/update_interaction.rs

```diff
@@ -0,0 +1,257 @@
+            interaction_group_id_1: Some(new_interaction_group_id_2.clone()),
+            interaction_group_id_2: None,
+        };
+
+        let result = service
+            .upsert_drug_interaction(service_provider.clone(), context.user_id.clone(), input)
+            .await
+            .unwrap();
+        assert_eq!(result, 1);
+
+        // Check that the item was added
+        let items = service.all_drug_interactions().await.unwrap();
+
+        let mut found = false;
+        for item in items.data {
+            if item.interaction_id == new_interaction_id {
+                assert_eq!(item.drugs.len(), 1);
+                assert_eq!(item.groups.len(), 1);
+                found = true;
+                break;
+            }
+        }
+        assert!(found);
+
+        // 4. Change to record with 2 drugs no groups
+        let input = UpsertDrugInteraction {
+            interaction_id: new_interaction_id.clone(),
+            name: new_interaction_id.clone(),
+            description: "4. Change to record with 2 drugs no groups".to_string(),
+            severity: dgraph::DrugInteractionSeverity::NothingExpected,
+            action: String::new(),
+            reference: String::new(),
+            drug_code_1: Some("cf00cc3f".to_string()),
+            drug_code_2: Some("c9896104".to_string()),
+            interaction_group_id_1: None,
+            interaction_group_id_2: None,
+        };
+
+        let result = service
+            .upsert_drug_interaction(service_provider.clone(), context.user_id.clone(), input)
+            .await
+            .unwrap();
+        assert_eq!(result, 1);
+
+        // Check that the item was added
+        let items = service.all_drug_interactions().await.unwrap();
+
+        let mut found = false;
+        for item in items.data {
+            if item.interaction_id == new_interaction_id {
+                assert_eq!(item.drugs.len(), 2);
+                assert_eq!(item.groups.len(), 0);
+                found = true;
+                break;
+            }
+        }
+        assert!(found);
+
+        // Delete the newly created interaction
+        let _result = service
+            .delete_drug_interaction(
+                service_provider.clone(),
+                context.user_id.clone(),
+                new_interaction_id.clone(),
+            )
+            .await
+            .unwrap();
+
+        // Delete the newly created groups
+        let _result = service
+            .delete_drug_interaction_group(
+                service_provider.clone(),
+                context.user_id.clone(),
+                new_interaction_group_id_1.clone(),
+            )
+            .await
+            .unwrap();
+
+        let _result = service
+            .delete_drug_interaction_group(
+                service_provider.clone(),
+                context.user_id.clone(),
+                new_interaction_group_id_2.clone(),
+            )
+            .await
+            .unwrap();
+    }
+}
```

Comment:
Might be good to add tests for providing i.e. 2 drugs and a group and also for i.e. 1 drug and no groups?

Reply:
[cd81661](https://github.com/msupply-foundation/unified-codes/pull/574/commits/cd816611590f8e2553a7be79aeabae6c66caede7)